app/scripts/filters/resources.js

```diff
@@ -529,8 +529,8 @@ angular.module('openshiftConsole')
   })
   .filter('isIncompleteBuild', function(ageLessThanFilter) {
     return function(build) {
-      if (!build || !build.status || !build.status.phase) {
-        return false;
+      if (!build || !build.status || !build.status.phase || !build.status.completionTimestamp) {
+        return true;
```

Comment:
part of this change is risky because this filter is used in so many places and may be expecting the behavior that a null/undefined build doesn't return true here.  I think everything except the `!build` check is ok to return true.  I would still return false for `!build`

If we actually have an issue in the build chart specifically with null / undefined builds, then we should fix the `completeBuilds` function in builds.js to specifically exclude those.

Reply:
Yeah, that makes sense. I didn't consider that this could cause null/undefined references to be returned. The problem wasn't that the builds are null/undefined, but that the `completionTimestamp` property was undefined. My train of thought was that this function should return true if that property is not defined (if it doesn't have a completionTimestamp then it's not complete). Maybe I should just add another check for that and leave the first conditional as it was?

Reply:
thats probably the safest change